### PR TITLE
Migrate mir_build diagnostics 2 of 3

### PR DESCRIPTION
**Pull request by mejrs**
The first three commits are fairly boring, however I've made some changes to the output of the match checking diagnostics.

### CODE REVIEW

**Comment by oli-obk** [comment]
I'm on vacation, I will review in 2-3 weeks

**Comment by mejrs** [comment]
No hurry :) Enjoy your holidays.

File: src/test/ui/pattern/usefulness/integer-ranges/overlapping_range_endpoints.stderr [hunk 1]
```diff
@@ -59,7 +59,6 @@ error: multiple patterns overlap on their endpoints
 LL |         0..=10 => {}
    |         ------ this range overlaps on `10_u8`...
 LL |         20..=30 => {}
-   |         ------- this range overlaps on `20_u8`...
```
Comment (oli-obk) [inline src/test/ui/pattern/usefulness/integer-ranges/overlapping_range_endpoints.stderr:62]: this information is lost

Comment (mejrs) [reply src/test/ui/pattern/usefulness/integer-ranges/overlapping_range_endpoints.stderr]: Fixed, although it required a hack I'm not proud of....

**Comment by oli-obk** [comment]
I think it's good to show these examples where we need hacks in the code, so we'll be able to address them if they are common enough.

@bors r+

**Comment by dtolnay** [comment]
Rebased to fix conflict with #106458, since this is showing not mergeable in bors.
https://bors.rust-lang.org/queue/rust
@bors r=oli-obk

**Pull request merged**